CPP/lists/LROrderedLinkedListSingle.h

```diff
@@ -333,6 +328,41 @@ class LROrderedLinkedListSingle {
     }
 
 
+    /**
+    * Clear the set.
+    *
+    * Progress Condition: Blocking
+    */
+    void clear() {
+        Node* node = nullptr;
+        Node* next = nullptr;
+
+        std::lock_guard<std::mutex> lock(_writersMutex);
+        if (_leftRight.load() == READS_ON_LEFT) {
+            _head->nextRight = _tail;
+            _leftRight.store(READS_ON_RIGHT);
+            toggleVersionAndWait();
+            node = _head->nextLeft;
+            while (node != _tail) {
+                next = node->nextLeft;
+                delete node;
+                node = next;
+            }
+        }
+        else {
+            _head->nextLeft = _tail;
+            _leftRight.store(READS_ON_LEFT);
+            toggleVersionAndWait();
+            node = _head->nextRight;
+            while (node != _tail) {
+                next = node->nextRight;
+                delete node;
+                node = next;
+            }
+        }
```

Comment:
and 
_head->nextRight = _tail;
is missing here, right?

Reply:
Yeah, you are right. I miss that.